### PR TITLE
feat: Introduce checkbox component

### PR DESCRIPTION
**Pull request by tatianainama**
Closes CP-2573

This PR introduces new Checkbox component using `@react-aria/checkbox`

[Figma Component](https://www.figma.com/file/vej9BUb6WbLxN9SpOK0Uqn/Prisma-Lens---Product?node-id=1728%3A1941)

<details>
<summary> Screenshot </summary>

image

</details>

### CODE REVIEW

**Comment by tatianainama** [comment]
Getting the following error while building Lens:

```
/src/lib.ts → ./dist/lib.js, ./dist/lib.esm.js...
[!] (plugin rpt2) Error: /components/menu/Menu.tsx(76,5): semantic error TS2345: Argument of type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' is not assignable to parameter of type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
  Property 'setOpen' is missing in type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' but required in type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
src/components/menu/Menu.tsx
```

I'm guessing this is related to the work done in previous PRs, isn't it?

File: tailwind.config.js [hunk 1]
```diff
@@ -72,6 +73,10 @@ module.exports = {
           "25%": { opacity: "1" },
           "100%": { transform: "translateY(0px)", opacity: "1" },
         },
+        ripple: {
+          from: { transform: "scale(0)" },
+          to: { transform: "scale(4)", opacity: "0" },
+        },
```
Comment (luanvdw) [inline tailwind.config.js:79]: This was a really nice touch!

File: src/components/checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      ref
+    )
+    const { focusProps, isFocusVisible } = useFocusRing()
+
+    return (
+      <label
+        className={cn("flex items-center", {
+          "flex-row-reverse": props.labelPosition === "left",
+          "cursor-pointer": !props.isReadOnly && !props.isDisabled,
+        })}
```
Comment (luanvdw) [inline src/components/checkbox/Checkbox.tsx:52]:
```suggestion
"cursor-not-allowed": props.isReadOnly || props.isDisabled,
        })}
```
Can we add a `not-allowed` cursor state for `isReadOnly` && `isDisabled `

**Comment by sampolahtinen** [comment]
>

**Comment by sampolahtinen** [comment]
> Getting the following error while building Lens:
> 
> ```
> /src/lib.ts → ./dist/lib.js, ./dist/lib.esm.js...
> [!] (plugin rpt2) Error: /components/menu/Menu.tsx(76,5): semantic error TS2345: Argument of type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' is not assignable to parameter of type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
>   Property 'setOpen' is missing in type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' but required in type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
> src/components/menu/Menu.tsx
> ```
> 
> I'm guessing this is related to the work done in previous PRs, isn't it?

This error is so mysterious. Here is some information I [shared in slack](https://prisma-company.slack.com/archives/G01KX8T24HK/p1662121569009119) 

So this error has something unknown to do with `yarn.lock`

> Getting the following error while building Lens:
> 
> ```
> /src/lib.ts → ./dist/lib.js, ./dist/lib.esm.js...
> [!] (plugin rpt2) Error: /components/menu/Menu.tsx(76,5): semantic error TS2345: Argument of type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' is not assignable to parameter of type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
>   Property 'setOpen' is missing in type 'import("/node_modules/@react-stately/menu/dist/types").MenuTriggerState' but required in type 'import("/node_modules/@react-aria/combobox/node_modules/@react-stately/menu/dist/types").MenuTriggerState'.
> src/components/menu/Menu.tsx
> ```
> 
> I'm guessing this is related to the work done in previous PRs, isn't it?

Argh this error is so mysterious. I shared some information [here](https://prisma-company.slack.com/archives/G01KX8T24HK/p1662121569009119)

It is related to `yarn.lock` file in some mysterious way. Let met quickly checkout your branch and see if this happens to me

**Comment by tatianainama** [comment]
> Argh this error is so mysterious. I shared some information [here](https://prisma-company.slack.com/archives/G01KX8T24HK/p1662121569009119)
> 
> It is related to `yarn.lock` file in some mysterious way. Let met quickly checkout your branch and see if this happens to me

I knew I read about this somewhere, thanks for the link!
So this might be because I added a new dependency, `@react-aria/checkbox` and updated the `yarn.lock` 🤔

**Comment by sampolahtinen** [comment]
> > Argh this error is so mysterious. I shared some information [here](https://prisma-company.slack.com/archives/G01KX8T24HK/p1662121569009119)
> > It is related to `yarn.lock` file in some mysterious way. Let met quickly checkout your branch and see if this happens to me
> 
> I knew I read about this somewhere, thanks for the link! So this might be because I added a new dependency, `@react-aria/checkbox` and updated the `yarn.lock` 🤔

Yeh exactly `it could be`. This is annoyingly flaky at the moment.

So I tried building the project using your branch and for me it works:

image

**Comment by sampolahtinen** [comment]
> > Argh this error is so mysterious. I shared some information [here](https://prisma-company.slack.com/archives/G01KX8T24HK/p1662121569009119)
> > It is related to `yarn.lock` file in some mysterious way. Let met quickly checkout your branch and see if this happens to me
> 
> I knew I read about this somewhere, thanks for the link! So this might be because I added a new dependency, `@react-aria/checkbox` and updated the `yarn.lock` 🤔

Yeh exactly `it could be`. This is annoyingly flaky at the moment.

So I tried building the project using your branch and for me it works:

image

**Comment by tatianainama** [comment]
@sampolahtinen
removing `node_modules` and installing everything again fixes the issue (as always lol)

**Comment by luanvdw** [comment]
> 

🙌  I re-deployed the preview link (without cache) and that solved the build on Netlify as well.

**Comment by luanvdw** [comment]
> @sampolahtinen removing `node_modules` and installing everything again fixes the issue (as always lol)

🙌  I re-deployed the preview link (without cache) and that solved the build on Netlify as well.

**Comment by sampolahtinen** [comment]
> @sampolahtinen removing `node_modules` and installing everything again fixes the issue (as always lol)

Classic. :D thats great! So I think something like this is happening:
1. The node_modules produced from unstable lens version includes different types some aria packages
2. And even if installing with different `yarn.lock` file, these packages and types do not get updated.
3. Thus wiping node_modules, installing clean with working `yarn.lock` fixes type mappings etc

**Comment by sampolahtinen** [comment]
> > @sampolahtinen removing `node_modules` and installing everything again fixes the issue (as always lol)
> 
> 🙌 I re-deployed the preview link (without cache) and that solved the build on Netlify as well.

This is something I cant wrap my head around. Why this keeps on occurring in Netlify. Why the fresh `node_modules` still throws errors in the next run...

File: src/components/checkbox/Checkbox.tsx (hunk 2 above)
Comment (tatianainama) [reply src/components/checkbox/Checkbox.tsx:53]: That's a great idea!

File: src/components/checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  }
+)
+
+const Ripple = ({ bgColor }: { bgColor: string }) => {
+  const rippleRef = useRef<HTMLSpanElement>(null)
+
+  useEffect(() => {
+    if (rippleRef.current) {
+      rippleRef.current.classList.remove("animate-ripple")
+      rippleRef.current.offsetWidth
```
[the remark this answers is not in the document]
Comment (sampolahtinen) [reply src/components/checkbox/Checkbox.tsx:123]: Oh interesting! Lets add a short comment and its good :)

Found this, might be interesting  to other as well https://gist.github.com/paulirish/5d52fb081b3570c81e3a

**Comment by sampolahtinen** [comment]
Thanks for the comment! LGTM! :)

**Review by sampolahtinen — APPROVED** [review]

**Pull request merged**